python/python/lance/dataset.py

```diff
@@ -839,6 +839,7 @@ def create_index(
         ivf_centroids: Optional[Union[np.ndarray, pa.FixedSizeListArray]] = None,
         num_sub_vectors: Optional[int] = None,
         accelerator: Optional[Union[str, "torch.Device"]] = None,
+        index_cache_size: Optional[int] = None,
```

Comment:
I wonder if it would be better to change `create_index` so that it modifies the dataset in-place instead of returning a new dataset.  Though that would be a breaking change so perhaps that ship has sailed.

Reply:
I was intending to do that at some point. I had done this on the Rust side earlier since it was a source of bugs: https://github.com/lancedb/lance/pull/1118

Reply:
That does seem like the better design to me.